staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -374,10 +394,27 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return nil
 		}
 
-		return o.PrintObj(outputObj, o.Out)
+		printer, err := o.ToPrinter(dataChangeMsg)
+		if err != nil {
+			return err
+		}
+
+		return printer.PrintObj(outputObj, o.Out)
 	})
 }
 
+// updateDataChangeMsg gets final operation msg
+func updateDataChangeMsg(oldObj []byte, newObj []byte, overwrite bool) string {
+	msg := MsgNotAnnotated
+	if !reflect.DeepEqual(oldObj, newObj) {
```

Comment:
Same question with above; isn't there other way without using reflection?

Reply:
Update with `cmp.Equal` to replace reflection, PTAL thanks.